testdata/p4_16_samples/structured-annotation.p4

```diff
@@ -0,0 +1,13 @@
+#define TEXT_CONST "hello"
+#define NUM_CONST 6
+
+@number[2]
+@empty[]
+@kv[k1=1,k2=2]
+@number(3)
```

Comment:
This program has `number` as both a structured and an unstructured annotation on the control `c`. This should be an error according to the intent of the proposed spec change p4lang/p4-spec#796.

Reply:
@liujed The intent of the spec change was to disallow  duplicate use of names  on a given *structured* annotation. Unstructured annotations have no such restriction. So `@number[2]` and `@number(3)` should be considered as non-colliding. By that reasoning, it would be OK to have another unstructured annotation in this program `number(4)` but not OK to add another structured annotation `number[5]`

Reply:
This is how the implementation was, but your last comment made it look like it's wrong, so I just changed it in my last PR.

Reply:
@chrispsommers, as Mihai says, your comment here suggests otherwise: https://github.com/p4lang/p4-spec/pull/796#issuecomment-598002044